### PR TITLE
Vcf parquet partitioning

### PR DESCRIPTION
**Pull request by IvoTod**

### CODE REVIEW

**Comment by IvoTod** [comment]
[Custom parquet dataset partitioning tools](https://trello.com/c/MHZaheGV/1298-custom-parquet-dataset-partitioning-tools)

File: dae/dae/backends/impala/parquet_io.py [hunk 1]
```diff
@@ -59,14 +59,51 @@ def __len__(self):
...
+        self.rows = rows
+
+    def _write_table(self):
+        print('WRITING')
+        self._writer.write_table(self._data.build_table())
+
+    def data_append(self, attr_name, value):
+        self._data.data_append(attr_name, value)
+
+    def write_if_overflowing(self):
```
Comment (lchorbadjiev) [inline dae/dae/backends/impala/parquet_io.py]: if the writer is 'automatic' probably we should put this check inside data_append

File: dae/dae/backends/tests/test_custom_parquet_writing.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import os
+import time
+from dae.backends.impala.parquet_io import VariantsParquetWriter
+
+
+def test_custom_parquet_writing(vcf_variants_loader):
```
[the remark this answers is not in the document]
Comment (lchorbadjiev) [reply dae/dae/backends/tests/test_custom_parquet_writing.py]: an also we need to check at least how many variants are stored in the dataset

File: dae/dae/backends/impala/parquet_io.py [hunk 3]
```diff
@@ -59,29 +61,103 @@ def __len__(self):
...
+            frequency_bin = 3
+
+        return frequency_bin
+
+    def evaluate_variant_filename(self, family_variant):
+        bins = (self._evaluate_region_bin(family_variant),)
+        filename = f'variants_region_bin_{bins[0]}'
+        if self.family_bin_size > 0:
+            bins += (self._evaluate_family_bin(family_variant),)
+            filename += f'_family_bin_{bins[len(bins)-1]}'
```
Comment (lchorbadjiev) [inline dae/dae/backends/impala/parquet_io.py]: don't get the just calculated bin using confusing bins indexing... store it in a variable, use it in the format and append it to the bins

File: dae/dae/backends/impala/parquet_io.py [hunk 4]
```diff
@@ -59,29 +61,103 @@ def __len__(self):
...
+        if len(self.coding_effect_types) > 0:
+            bins += (self._evaluate_coding_bin(family_variant),)
+            filename += f'_coding_bin_{bins[len(bins)-1]}'
+        if self.rare_boundary > 0:
+            bins += (self._evaluate_frequency_bin(family_variant),)
+            filename += f'_frequency_bin_{bins[len(bins)-1]}'
+        filepath = os.path.join('', *tuple(map(str, bins)))
```
Comment (lchorbadjiev) [inline dae/dae/backends/impala/parquet_io.py]: partition directory name should include the partition column name...

**Pull request merged**